keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md

```diff
@@ -0,0 +1,919 @@
+        // Expression must produce a string value.
+        // Possible prefixes based on the config:
+        //     (1) if userName.prefix = "-", no prefix will be added to the username
+        //     (2) if userName.prefix = "" and userName.claim != "email", prefix will be "<issuer.url>#"
+        //     (3) if userName.expression is set instead, result of expression is used as-is without any implicit prefix
+        // (1) and (2) ensure backward compatibility with the --oidc-username-claim and --oidc-username-prefix flags
+        Username PrefixedClaimOrExpression `json:"username"`
+        // Groups represents an option for the groups attribute.
+        // Claim must be a string or string array claim.
+        // Expression must produce a string or string array value.
+        // "", [], missing, and null values are treated as having no groups.
+        // TODO: investigate if you could make a single expression to construct groups from multiple claims. If not, maybe []PrefixedClaimOrExpression?
+        // +optional
+        Groups PrefixedClaimOrExpression `json:"groups,omitempty"`
+        // UID represents an option for the uid attribute.
+        // Claim must be a singular string claim.
+        // Expression must produce a string value.
+        // TODO: this is net new, should it just be expression?
+        // +optional
+        UID ClaimOrExpression `json:"uid,omitempty"`
+        // Extra represents an option for the extra attribute.
+        //
+        // TODO: examples for this?
+        //
+        // # known key, value from claim
+        // - key: "example.com/myextrakey"
+        //   value:
+        //     claim: "hd"
+        //
+        // # known key, value constructed by expression
+        // - key: "example.com/myextrakey"
+        //   value:
+        //     expression: claims.someclaim+":"+claims.someclaim
+        //
+        // # calculated key/value pairs? CEL returns [{key,value}, {key,[value, value...]}, ...] and we aggregate?
+        // TODO: ask joe/cici about CEL constructing/returning complex types
+        //
+        // +optional
+        Extra []ExtraMapping `json:"extra,omitempty"`
+    }
+
+    type ExtraMapping struct {
+        // Key is a CEL expression to extract extra attribute key.
+        // Claim must be a singular string claim.
+        // Expression must produce a string value.
+        // "" and null values are treated as the extra mapping not being present.
+        Key string `json:"key"`
+        // Value is a CEL expression to extract extra attribute value.
+        // Claim must be a string or string array claim.
+        // Expression must produce a string or string array value.
+        // "", [], and null values are treated as the extra mapping not being present.
+        Value ClaimOrExpression `json:"value"`
```

Comment:
limit this to inline `valueExpression` since this is new (see above examples for how to express pulling values directly from claims)